apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/create/hooks/use-create-ecosystem.ts

```diff
@@ -0,0 +1,97 @@
+import { thirdwebClient } from "@/constants/client";
+import {
+  DASHBOARD_THIRDWEB_CLIENT_ID,
+  IPFS_GATEWAY_URL,
+} from "@/constants/env";
+import { useLoggedInUser } from "@3rdweb-sdk/react/hooks/useLoggedInUser";
+import {
+  type UseMutationOptions,
+  useMutation,
+  useQueryClient,
+} from "@tanstack/react-query";
+import { upload } from "thirdweb/storage";
+
+export type CreateEcosystemParams = {
+  slug: string;
+  name: string;
+  logo: File;
+  permission: "PARTNER_WHITELIST" | "ANYONE";
+};
+
+export function useCreateEcosystem(
+  options?: Omit<
+    UseMutationOptions<string, unknown, CreateEcosystemParams>,
+    "mutationFn"
+  >,
+) {
+  const { onSuccess, ...queryOptions } = options ?? {};
+  const { isLoggedIn } = useLoggedInUser();
+  const queryClient = useQueryClient();
+
+  const { mutateAsync: createEcosystem, isLoading } = useMutation({
+    // Returns the created ecosystem slug
+    mutationFn: async (params: CreateEcosystemParams): Promise<string> => {
+      if (!isLoggedIn) {
+        throw new Error("Please login to create an ecosystem");
+      }
+
+      const imageUri = await upload({
+        client: thirdwebClient,
+        files: [params.logo],
+      });
+
+      // Create the IPFS
+      const imageUrl = IPFS_GATEWAY_URL.replace(
+        "{clientId}",
+        DASHBOARD_THIRDWEB_CLIENT_ID,
+      )
+        .replace("{cid}", imageUri.split("/")[2] ?? "")
+        .replace("{path}", imageUri.split("/")[3] ?? "");
```

Comment:
this might be bad? Are we writing the dashboard client id into the "imageUrl?

instead simply write the `ipfs://<uri>` into the payload, any downstream renderer of the image Url has to use *their* clientId to resolve it later